submission/distributor_test.go

```diff
@@ -364,30 +345,21 @@ func TestDistributorAddPreChain(t *testing.T) {
 			ctx, cancel := context.WithTimeout(context.Background(), time.Second)
 			defer cancel()
 
-			go dist.Run(ctx)
-			// First Log refresh expected.
-			<-ctx.Done()
+			dist.Run(ctx)
 
 			scts, err := dist.AddPreChain(context.Background(), tc.rawChain)
 			if gotErr := (err != nil); gotErr != tc.wantErr {
 				t.Errorf("Expected to get errors is %v while actually getting errors is %v", tc.wantErr, gotErr)
 			}
 
-			var gotNum, wantNum int
-			if scts != nil {
-				gotNum = len(scts)
-			}
-			if tc.scts != nil {
-				wantNum = len(tc.scts)
-			}
-			if gotNum != wantNum {
-				t.Errorf("Expected to get %d SCTs on AddPreChain request, got %d", wantNum, gotNum)
+			if got, want := len(scts), len(tc.scts); got != want {
+				t.Errorf("Expected to get %d SCTs on AddPreChain request, got %d", want, got)
 			}
 			gotMap := getSCTMap(tc.scts)
 			for _, asct := range scts {
 				if wantedSCT, ok := gotMap[asct.LogURL]; !ok {
-					t.Errorf("Expected to get SCT from Log %q, but got none", asct.LogURL)
-				} else if !reflect.DeepEqual(asct, wantedSCT) {
+					t.Errorf("dist.AddPreChain() = (_, %v), want err? %t", err, tc.wantErr)
+				} else if diff := cmp.Diff(asct, wantedSCT); diff != "" {
 					t.Errorf("Got unexpected SCT for Log %q", asct.LogURL)
```

Comment:
Should include the diff in the error message, otherwise this is no better than `reflect.DeepEqual()`.